### PR TITLE
basicauthproxy: Add Basicauth http proxy example

### PR DESCRIPTION
**Pull request by hajhatten**

### CODE REVIEW

File: basicauthproxy/Caddyfile [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+# If you want to use more than one user, just append more rows with the next user
+# basicauth / username1 password1
+# basicauth / username2 password2
+
+basicauth / username password
+
+# Proxy to localhost port 8080
+# If multiple backends are used, just specify them comma separated e.g.
+# proxy / localhost:8080, localhost:8081, 192.168.99.100:8083
```
Comment (mholt) [inline basicauthproxy/Caddyfile]: The comma separated advice is not correct; these should be space-separated.

File: basicauthproxy/README.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Basic Auth HTTP proxy with Caddy server
+
+In this document you will learn on how to set up Caddy server as a basicauth proxy in front of a http server.
```
Comment (mholt) [inline basicauthproxy/README.md]: "set up Caddy as ..."

File: basicauthproxy/README.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+For more information, see the [http.basicauth](https://caddyserver.com/docs/basicauth) and [http.proxy](https://caddyserver.com/docs/proxysection) of the documentation.
```
Comment (mholt) [inline basicauthproxy/README.md]: Correct proxy link is `/docs/proxy`

File: basicauthproxy/README.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+#### Prerequisites
+
+* You have a http server running on localhost or an external resource
+* You have Caddy server installed, if not follow [Getting Started with Caddy](https://caddyserver.com/docs/getting-started)
+
+#### Start caddy server
```
Comment (mholt) [inline basicauthproxy/README.md]: "Start Caddy"

File: basicauthproxy/README.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
+# Basic Auth HTTP proxy with Caddy server
```
Comment (mholt) [inline basicauthproxy/README.md]: Call it "Caddy" - not "Caddy server". Thanks :)

**Comment by hajhatten** [comment]
Thanks mholt, you should have another look on the minio readme after these corrections.

**Pull request merged**